### PR TITLE
In 1065 maintenance 09 2024

### PR DESCRIPTION
**Pull request by jonavellecuerdo**
### Purpose and background context

Update app according to [DataEng Application Maintenance](https://mitlibraries.atlassian.net/wiki/spaces/IN/pages/3445948428/DataEng+Application+Maintenance).

The major updates for this work include:
* Bumping Python version from 3.11 to 3.12
* Creating a `Config` class to support dynamic access to environment variables
* Updating `AlmaClient` to use `Config` class (and deprecating `config.load_alma_config`
   **Note:** I opted for minimal updates and didn't convert the class to `attrs` hence the use of `@property` methods for the class attributes that relied on access to env variables via `Config`. 
* Updating `configure_logger` to use `verbose` boolean flag
   * This resulted in deprecation of `-l/--log-level` CLI option and updating CLI to use `-v/--verbose` CLI option instead.
   **Note:** The ECS task definition in **Prod** currently sets `LOG_LEVEL`. This will not cause an issue as `Config().LOG_LEVEL` is not called anywhere in the application, but nothing that we'd need to ask InfraEng to update the Terraform docs to remove this env variable.

### How can a reviewer manually see the effects of these changes?
Run `make test` and verify all unit tests are passing.

### Includes new or updated dependencies?
YES

### Changes expectations for external applications?
NO

### What are the relevant tickets?
- https://mitlibraries.atlassian.net/browse/IN-1065

### Developer
- [X] All new ENV is documented in README
- [ ] All new ENV has been added to staging and production environments
- [X] All related Jira tickets are linked in commit message(s)
- [X] Stakeholder approval has been confirmed (or is not needed)

### Code Reviewer(s)
- [ ] The commit message is clear and follows our guidelines (not just this PR message)
- [ ] There are appropriate tests covering any new functionality
- [ ] The provided documentation is sufficient for understanding any new functionality introduced
- [ ] Any manual tests have been performed and verified
- [ ] New dependencies are appropriate or there were no changes

### CODE REVIEW

**Review by ghukill — APPROVED** [review]
Changes look good to me!  Thanks for the detailed PR, easy to follow along the changes.

I'm happy to approve -- tests and linting passes locally -- but @jonavellecuerdo it might be worth tagging Adam S. as a reviewer on this PR as well.  He's likely in a better condition to perform any manual tests (if possible).

**Comment by ehanson8** [comment]
> Changes look good to me! Thanks for the detailed PR, easy to follow along the changes.
> 
> I'm happy to approve -- tests and linting passes locally -- but @jonavellecuerdo it might be worth tagging Adam S. as a reviewer on this PR as well. He's likely in a better condition to perform any manual tests (if possible).

Adam S. doesn't do anything with this one, just SAP invoices. I still respond to all of the tickets related to this app

File: ccslips/cli.py [hunk 1]
```diff
@@ -36,48 +37,51 @@
     "--date",
     help=(
         "Optional date of exports to process, in 'YYYY-MM-DD' format. Defaults to "
-        "yesterday's date if not provided."
+        "two (2) days before the date the application is run."
```
Comment (ehanson8) [inline ccslips/cli.py:40]: Good catch on all of these, I should have updated them when I made the change

File: ccslips/config.py [hunk 2]
```diff
@@ -1,27 +1,48 @@
...
 import sentry_sdk
 
 
-def configure_logger(logger: logging.Logger, log_level_string: str) -> str:
-    if log_level_string.upper() not in logging.getLevelNamesMapping():
-        message = f"'{log_level_string}' is not a valid Python logging level"
-        raise ValueError(message)
-    log_level = logging.getLevelName(log_level_string.upper())
-    if log_level < 20:  # noqa: PLR2004
+class Config:
```
Comment (ehanson8) [inline ccslips/config.py:8]: Good change, I appreciate the standardization around this approach!

**Review by ehanson8 — APPROVED** [review]
Great extra enhancements beyond the usual dependency and Python updates!

**Pull request merged**